src/python/grpcio_tests/tests/protoc_plugin/_python_plugin_test.py

```diff
@@ -503,5 +504,118 @@ def half_duplex_request_iterator():
         service.server.stop(None)
 
 
+@unittest.skipIf(sys.version_info[0] < 3, "Unsupported on Python 2.")
+class SimpleStubsPluginTest(unittest.TestCase):
+    servicer_methods = _ServicerMethods()
+
+    class Servicer(service_pb2_grpc.TestServiceServicer):
+
+        def UnaryCall(self, request, context):
+            return SimpleStubsPluginTest.servicer_methods.UnaryCall(
+                request, context)
+
+        def StreamingOutputCall(self, request, context):
+            return SimpleStubsPluginTest.servicer_methods.StreamingOutputCall(
+                request, context)
+
+        def StreamingInputCall(self, request_iterator, context):
+            return SimpleStubsPluginTest.servicer_methods.StreamingInputCall(
+                request_iterator, context)
+
+        def FullDuplexCall(self, request_iterator, context):
+            return SimpleStubsPluginTest.servicer_methods.FullDuplexCall(
+                request_iterator, context)
+
+        def HalfDuplexCall(self, request_iterator, context):
+            return SimpleStubsPluginTest.servicer_methods.HalfDuplexCall(
+                request_iterator, context)
+
+    def setUp(self):
+        super(SimpleStubsPluginTest, self).setUp()
+        self._server = test_common.test_server()
+        service_pb2_grpc.add_TestServiceServicer_to_server(
+            self.Servicer(), self._server)
+        self._port = self._server.add_insecure_port('[::]:0')
+        self._server.start()
+        self._target = 'localhost:{}'.format(self._port)
+
+    def tearDown(self):
+        self._server.stop(None)
+        super(SimpleStubsPluginTest, self).tearDown()
+
+    def testUnaryCall(self):
+        request = request_pb2.SimpleRequest(response_size=13)
+        response = service_pb2_grpc.TestService.UnaryCall(
+            request,
+            self._target,
+            channel_credentials=grpc.experimental.insecure_channel_credentials(
+            ),
+            wait_for_ready=True)
```

Comment:
nit: Can we simplify the `grpc.experimental.insecure_channel_credentials` design?
nit: Is `wait_for_ready=True` necessary for tests to pass on Windows? If the underlying channel is pooled, can we warm up the channel object earlier to prevent explicitly setting this flag?

Reply:
That's rather vague. How do you mean? Were you expecting insecure to be default? Were you expecting the name to be shorter? FWIW, the default is currently actually `LocalCredentials`, which I was originally using here. But I ran into  https://github.com/grpc/grpc/issues/22257, which is why I'm explicitly specifying here. Eric and Sanjay both have weighed in that explicit is better than implicit for security reasons.

`wait_for_ready` is necessary in general to make this test not flaky. Otherwise, there's a race condition between the server thread coming ready and the main test thread sending the request. I don't think relying on the implementation detail that channels are pooled is a good idea. `wait_for_ready` is already a standard solution for this. However, we *could* flip the default to `True` for `wait_for_ready`, but we should probably have a bit more of a conversation about that first.

Reply:
The `LocalCredentials` issue is not meant to be portable, maybe we should consider other default value here? Should we use `insecure_channel` by default? It seems least surprising to our users.

This issue seems more or less test-specific. We could try to use `channel_ready_future` in `setUp` to ensure the server is ready, or other mechanism. This can be in a separate PR.

About flipping the default value for `wait_for_ready`, in usual scenario, gRPC Python are connecting to some public services, where fail-fast helps users to debug their configuration error. So, I think it should stay `False`.

My intent is to make this test cleaner. Ideally, if we share this test case to users, they can immediately agree that the new API is much more simpler.

Reply:
I definitely agree that `LocalCredentials` cannot remain the default.

Everyone I have talked to about it agrees that is not acceptable for an API to be insecure by default. So `insecure_channel` is out as well.

You could use `ssl_credentials` as your default option. But that doesn't make sense inside of google3. You'd want `loas2` there instead.

You could do something more complicated, like the `CompositeCredentials` idea I proposed a while ago. But, even in a tiered solution like this, you'd need to pick between SSL and LOAS2.

In short, this is a complicated issue. It's one that I think deserves more attention, and will probably require convincing other parts of the team. For the moment, I'd like to remove the default credential, since it is always backwards compatible to add one in.

Reply:
> For the moment, I'd like to remove the default credential, since it is always backwards compatible to add one in.

Changing default value can be surprising to users. This might be in the scope of PR, but we might need a better solution.

Reply:
This is an unreleased experimental API that emits a warning when you use it. Removing the default value is definitely fair game. As I stated above, I would like to look into a default value for this feature, but let's not tie it to this PR.

Reply:
Temporarily removed the default and added a TODO. I'm looking into some longer term solutions to this problem.